Feature/yolo testnet

sambacha opened the pull request:
<!-- Thanks for sending a pull request! Please check out our contribution guidelines: -->
<!-- https://github.com/hyperledger/besu/blob/master/CONTRIBUTING.md -->

## PR description

## Fixed Issue(s)
<!-- Please link to fixed issue(s) here using format: fixes #<issue number> -->
<!-- Example: "fixes #2" -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/freight-trust/besu?pullRequest=1) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/freight-trust/besu?pullRequest=1) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: edwardmack<br/>:x: shemnon<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/freight-trust/besu?pullRequest=1) it.</sub>

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/freight-trust/besu?pullRequest=1) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/freight-trust/besu?pullRequest=1) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: edwardmack<br/>:x: shemnon<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/freight-trust/besu?pullRequest=1) it.</sub>

AbdelStark committed:
Fixed IndexOutOfBoundsException in DebugOperationTracer (#946)

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

yehia67 committed:
remove removeAllKeysUnless from interfaces and its implemntation (#936)


Signed-off-by: yehia67 <yehiatarek67@gmail.com>

EdJoJob committed:
Set the notifications webhook by URL to prevent false alerts (#952)

In the event that someone who has forked this repository sets up CircleCI integrations for it, we would like that builds on their master branch would not be able to send alerts to the Hyperledger Rocket Chat #besu-contributors channel. Here we move to having the notification web-hook be sent to an address set by the HUBOT_URL variable instead. This environment variable is set in the CircleCI web-app in the same place we set the variables for our secrets for other 3rd party app access.

It has been tested that an environment without the variable being set would cause no issues for the CircleCI build system either for other forked projects.

Signed-off-by: Edward Evans <edward.joshua.evans@gmail.com>

mark-terry committed:
Updated SimpleNodePermissioning smart contract to relax enodeURL matc… (#945)

Updated SimpleNodePermissioning smart contract in the acceptance tests to relax enodeURL matching to exclude the p2p port.

Signed-off-by: Mark Terry <mark.terry@consensys.net>

RatanRSur committed:
make ReturnStack into singleton class (#950)

Signed-off-by: Ratan Rai Sur <ratan.r.sur@gmail.com>

shemnon committed:
Revert "make ReturnStack into singleton class (#950)" (#962)

This reverts commit 26af974913266363eea994af52ffdb21a603881e.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

shemnon committed:
Performance: error message calculation (#955)

Only calculate the error message if an error occurs.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

shemnon committed:
Log all persisted blocks (#957)

Because of where we provide info log messages in the
BlockPropagationManager we don't log out of order blocks.  Moving to
task cleanup allows us to get the timing information too instead of
simply as part of the executeTask method

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

MadelineMurray committed:
Added upcoming upgrade section (#937)

* Added upcoming upgrade section

Signed-off-by: Madeline <madeline.murray@consensys.net>

* Edit

Signed-off-by: Madeline <madeline.murray@consensys.net>

* rework

Signed-off-by: Madeline <madeline.murray@consensys.net>

* rework

Signed-off-by: Madeline <madeline.murray@consensys.net>

MadelineMurray committed:
Added eth_estimateGas fix and logs query known issue (#965)

Signed-off-by: Madeline <madeline.murray@consensys.net>

matkt committed:
Throw error if full sync is requested when fast sync is incomplete (#958)

It was possible to run Besu on an incomplete fast-synced db with full sync enabled. This causes the node to stall out. This PR resolves this issue.

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

shemnon committed:
Eip-2537 - BLS12-381 precompiles (#964)

Implement BLS12-381 precompiles via the MatterLabs library.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

mark-terry committed:
Fixed bug where a permissioned IBFT2 network starts in stalled state.  (#969)

Signed-off-by: Mark Terry <mark.terry@consensys.net>

matkt committed:
remove Intrinsic gas exceeds gas limit known issue (#971)

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

AbdelStark committed:
[EIP-2124] Backward compatible fork id management implementation (#979)

* Handle zeros fork blocks when computing fork id.

- Updated `ForkIdManager`:
    - Created `ForkIDChecker` interface with `boolean check(ForkId forkId)` method.
    - Added a check in the constructor to use the proper `ForkIDChecker` based on the following conditions:
        - Accept all peers if the list of forks contains only `zeros`.
        - Use `EIP-2124` rules otherwise.

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

* Handle backward compatibility

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

matkt committed:
Update EIP-2315 implementation (enter only via jumpsub) (#941)

Update implementation of the EIP-2315 
- Execution of BEGINSUB causes exception and terminates execution.
- JUMPSUB sets the pc to location + 1 .
- In the edge case when BEGINSUB is the last instruction in code and this subroutine is jumped-to, the implementations should execute STOP .

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

AbdelStark committed:
Handle backward compatibility for EIP-2124 fork id management (#981)

* Handle backward compatibility for EIP-2124 fork id management

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

shemnon committed:
Enforce devp2p packet expiration (#978)

Don't respond to neighbors and ping packets that have an expiration
prior to the system's current time.

Addresses two tests in #975

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

macfarla committed:
[MINOR] Typo env (#985)

* typo

Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

shemnon committed:
update spdx exclusions (#988)

SPDX checks shouldn't be done against jmh generated files.  Make it
generic so that the gradle/intellij favoured location of generated files
will never be a problem. These are also covered by `.gitignore` already.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

lucassaldanha committed:
Add gate for multi-tenancy + onchain privacy groups (#990)

Signed-off-by: Lucas Saldanha <lucas.saldanha@consensys.net>

MadelineMurray committed:
Added 938 to changelog (#993)

Signed-off-by: Madeline <madeline.murray@consensys.net>

AbdelStark committed:
Add changelog entry for price bump mechanism. (#994)

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

AbdelStark committed:
Add timeout handler in JsonRpcHttpService. (#986)

* Add timeout handler in JsonRpcHttpService.

- Created `TimeoutOptions` class with fields:
    - *timeout*: the timeout in milliseconds.
    - *errorCode*: the http status code.
- Created `EthRpcTimeoutHandler`.
    -  Contains `handler` static method to create `Vertx Handler`. The process of the handler created is the following
        - retrieve the body as string.
        - parse it as a `JsonObject`.
        - cache the resulting `JsonObject` in the `RoutingContext`.
        - retrieve the RPC method name.
        - set a timer based on the configuration if the method name matches one entry in the configuration map.
        - call the next handler.

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

shemnon committed:
Rework PendingTransactions synchronization (#972)

Slight rework on some PendingTransaction synchronization
* sync on `prioritizedTransactions` instead of `pendingTransactions`
  * `pendingTransactions` is already a `ConcurrentHashMap`, which
    syncs on itself internally
  * not all accesses to `pendingTransactions` were synced
  * `prioritizedTransactions` is not exposed directly or indirectly,
    making it a better lock
* Improve synchronization in TransactionsForSenderInfo
  * `gaps` was returned without requiring synchronization
  * `transacitonInfos` was not thread safe.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

RatanRSur committed:
make mark/sweep prepare async with main thread (#947)

We cleared the mark storage upon starting the pruner in order to prevent the first sweep from being smaller than it needs to be.

 // Optimization for the case where the previous cycle was interrupted (like the node was shut 
 // down). If the previous cycle was interrupted, there will be marks in the mark storage from 
 // last time, causing the first sweep to be smaller than it needs to be. 
 clearMarks(); 

Unfortunately, it's currently called in prepare, which is synchronous with the main thread. This means that we can delay startup by on the order of 10 minutes if the storage was especially full.

This commit makes the pruner `start` asynchronous.

Signed-off-by: Ratan Rai Sur <ratan.r.sur@gmail.com>

mark-terry committed:
Changelog for PR #969. (#984)

* Changelog for PR #969.

Signed-off-by: Mark Terry <mark.terry@consensys.net>

* Tweak.

Signed-off-by: Mark Terry <mark.terry@consensys.net>

Co-authored-by: MadelineMurray <43356962+MadelineMurray@users.noreply.github.com>

shemnon committed:
Use seconds for Timeout Configuration (#997)

* Use seconds for timeout configuration
* Add units to variable names
* Change timeout to 5 minutes

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

MadelineMurray committed:
Updated known issues for 1.4.6  (#996)

* Updated known issues

Signed-off-by: Madeline <madeline.murray@consensys.net>

joshuafernandes committed:
1.4.6-RC1 release (#998)

Signed-off-by: Joshua Fernandes <joshua.fernandes@consensys.net>

joshuafernandes committed:
Prepare for version 1.4.7-SNAPSHOT (#999)

Signed-off-by: Joshua Fernandes <joshua.fernandes@consensys.net>

macfarla committed:
[MINOR] Fixed failing tests in account permissioning smart contract (#956)

* fixed tests that fail

Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

Co-authored-by: mark-terry <36909937+mark-terry@users.noreply.github.com>

ajsutton committed:
Revert "make mark/sweep prepare async with main thread (#947)" (#1001)

Suspected of making PrunerTest intermittent.

Signed-off-by: Adrian Sutton <adrian.sutton@consensys.net>

shemnon committed:
Improve Eth/64 compatibility (#1005)

Rollback a Eth64 change intended to improve compatibility in consortium network. It resulted in many nodes not connecting on named networks such as goerli and mordor.

* Revert "Handle backward compatibility for EIP-2124 fork id management (#981)"
  This reverts commit 1d8218ae
* Revert "[EIP-2124] Backward compatible fork id management implementation (#979)"
  This reverts commit cb2c67d7

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

RatanRSur committed:
make mark/sweep prepare async with main thread (this time without test intermittency) (#1004)

We cleared the mark storage upon starting the pruner in order to prevent the first sweep from being smaller than it needs to be.
```
 // Optimization for the case where the previous cycle was interrupted (like the node was shut 
 // down). If the previous cycle was interrupted, there will be marks in the mark storage from 
 // last time, causing the first sweep to be smaller than it needs to be. 
 clearMarks(); 
```
Unfortunately, it's currently called in prepare, which is synchronous with the main thread. This means that we can delay startup by on the order of 10 minutes if the storage was especially full.

This commit makes the pruner `start` asynchronous.

This was reverted once before due to test intermittency and this commit is the second attempt :P

Signed-off-by: Ratan Rai Sur <ratan.r.sur@gmail.com>

shemnon committed:
DevP2P expiration conformance (#1006)

DevP2P packets expiration fields should be in seconds past the epoch,
not milliseconds past the epoch.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

rain-on committed:
Repair intermittent EthGetWork test (#1008)

The EthGetWork.shouldReturnSuccessResponseWhenMining was exhibiting intermittent failures, as a mining node would occasionally report "No mining work available yet" - rather than the inputs required to be solved for the current block.

From investigation, it was found that Besu's EthHash miner has a small time window after completing a block, in which the target for the next block has not yet been computed (eg while collating transactions) - during this interval, Besu will validly report "No mining work available yet".

To overcome this, the acceptance test has been updated to poll Besu for upto 5 seconds, waiting for a valid work target to be provided - rather than executing a single-attempt (which may fall within the window specified above).

Signed-off-by: Trent Mohay <trent.mohay@consensys.net>

shemnon committed:
Improve stability of CI tests (#878)

* Use non-blocking randomness for acceptance tests
  This addresses entropy draining during unit tests.
* wait 30 seconds instead of 2 when killing AT processes
* mark NodeSmartContractPermissioningIbftStallAcceptanceTest as @Ignore since it has become reliably and specifically flakey.

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

edwardmack committed:
update getForks to ignore ClassicForkBlock param, update gotham number (#1014)

* update getForks to ignore classicforkblock param

Signed-off-by: Edward Mack <ed@edwardmack.com>

* update gotham block fork number

Signed-off-by: Edward Mack <ed@edwardmack.com>

* add to entry to CHANGELOG.md

Signed-off-by: Edward Mack <ed@edwardmack.com>

matkt committed:
Fix Ethereum classic mining issue (#1009)

Resolve an issue during mining related to the estimation of rewards on the Ethereum classic network.

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

matkt committed:
update changelog (#1017)

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

matkt committed:
EIP-2315 change gascosts and update opcode (#995)

Change gascosts

  - JUMPSUB -> 10 (high tier gas cost)
  - RETURNSUB -> 5 (low tier gas cost)
  - BEGINSUB -> 2 (base tier gas cost)

Update opcode 

  - 0x5c BEGINSUB
  - 0x5d RETURNSUB
  - 0x5e JUMPSUB

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

RatanRSur committed:
make build aware of docker memory limitations (#1015)

We are seeing the gradle daemon die when running the acceptance tests

```
The message received from the daemon indicates that the daemon has 
disappeared.
```

Starting from Java 10, UseContainerSupport is turned on by default, 
except in the case whereXmx is specified. I'll be honest in that I 
don't know why the Xmx of 1gb was specified in the first place, so this 
change is bad on the Chesterton's Fence front but if it has anything to 
do with limiting the memory usage on containers, removing it would 
probably be better.

Signed-off-by: Ratan Rai Sur <ratan.r.sur@gmail.com>

AbdelStark committed:
Reintroduce consortium network friendly EIP-2124 fork id management. (#1018)

* Handle backward compatible fork id.

Signed-off-by: Abdelhamid Bakhta <abdelhamid.bakhta@consensys.net>

schroedingerscode committed:
Accept Nested TOML Array for Default Value Decoding (#920)

* Handle nested TOML array to resolve TOML array decoding bug in certain plugin configs.
* Fix for picocli unable to determine if plugin options are multivalue.

Signed-off-by: Steven J Schroeder <steven.schroeder@consensys.net>

matkt committed:
Verify jumpsub destination validation (#1022)

Improved JUMP and JUMPSUB destination check. Avoid having to browse the code if the destination == code size

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

matkt committed:
init yolo testnet

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

matkt committed:
add yolo testnet genesis

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

matkt committed:
fix config issue

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>

matkt committed:
add test

Signed-off-by: Karim TAAM <karim.t2am@gmail.com>
